@here/harp-datasource-protocol/lib/TechniqueParams.ts

```diff
@@ -691,12 +691,32 @@ export interface MarkerTechniqueParams extends BaseTechniqueParams {
     wrappingMode?: DynamicProperty<"None" | "Character" | "Word">;
     /**
      * Text position regarding the baseline.
+     *
+     * @deprecated Use [[hPlacement]] attribute, which defines label text placement in exactly
+     * opposite way, text that were `Left` aligned are by deduction `Right` placed relative to
+     * the label icon.
      */
     hAlignment?: DynamicProperty<"Left" | "Center" | "Right">;
     /**
      * Text position inside a line.
+     *
+     * @deprecated Use [[vPlacement]] attribute instead.
      */
     vAlignment?: DynamicProperty<"Above" | "Center" | "Below">;
+    /**
+     * Text position regarding the baseline.
+     *
+     * @deprecated Use [[hPlacement]] attribute, which defines label text placement in exactly
+     * opposite way, text that were `Left` aligned are by deduction `Right` placed relative to
+     * the label icon.
+     */
+    hPlacement?: DynamicProperty<"Left" | "Center" | "Right">;
```

Comment:
Not sure how harp-native is doing this but if we want to support multiple placements in the future this would become an array and it would be independent from `vPlacement`. Might be better to combine both placements into one property like
```
placement: {
    horizontal: "Left" | "Center" | "Right",
    vertical: "Top" | "Center" | "Bottom"}
```

Reply:
not sure how well this works with `DynamicProperties` though.

Reply:
@ninok 
In case of dynamic properties, we cannot use "plain old array", but we need to use [ "array", ... ] expression and we're fine. It will be just a little more verbose.
I.e
```
"placement": ["array", { a: 1, b: 2 }, { a: 2, c: 2) ...]
```
instead of intuitive:
```
placement: [{ a: 1, b: 2 }, { a: 2, c: 2)]
```

But, i think this patch is far from having support for array of placement options.

Reply:
okay, cool.

> But, i think this patch is far from having support for array of placement options.

I know but I just want to make sure that the style will not change again when we introduce array of placements. So I propose that the placement is a literal now, i.e.:
```
placement: ["literal", { v: "Center", h: "Center" }]
```
and in the future we could extend it to support both, literals and arrays as you proposed

Reply:
Guys, nice to see the strong discussion here, but I guess the ticket description was in totally different area and now AC have changed significantly, probably this entire PR will be outdated, but need to wait for formal AC definition. I have also kind of "make it all" proposal in the Jira ticket comments:
```
placement: [
                "case",
                        ["==", ["get", "readDirection"], "left-right"],
                        ["array", "TR", "TL", "CC" ],
                        ["==", ["get", "readDirection"], "right-left"],
                        ["array", "TL", "TR", "CC" ]
                 ...
             ]
```